encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/persistence/dao/EncryptedLogDao.kt

```diff
@@ -0,0 +1,66 @@
+package com.automattic.encryptedlogging.persistence.dao
+
+import androidx.room.Dao
+import androidx.room.Delete
+import androidx.room.Insert
+import androidx.room.Query
+import androidx.room.Upsert
+import com.automattic.encryptedlogging.model.encryptedlogging.EncryptedLog
+import com.automattic.encryptedlogging.model.encryptedlogging.EncryptedLogModel
+
+@Dao
+internal abstract class EncryptedLogDao {
+    @Query("""
+        SELECT * FROM EncryptedLogEntity 
+        WHERE uuid = :uuid
+    """)
+    internal abstract suspend fun getEncryptedLog(uuid: String): EncryptedLogModel?
+
+    @Query("""
+        SELECT * FROM EncryptedLogEntity 
+        WHERE uploadStateDbValue IN (:uploadStates) 
+        ORDER BY uploadStateDbValue ASC, id ASC 
+        LIMIT 1
+    """)
+    internal abstract suspend fun getEncryptedLog(uploadStates: List<Int>): EncryptedLogModel?
+
+    @Query("""
+        SELECT * FROM EncryptedLogEntity 
+        WHERE uploadStateDbValue = :uploadState
+    """)
+    internal abstract suspend fun getEncryptedLogs(uploadState: Int): List<EncryptedLogModel>
+
+    @Query("""
+        SELECT COUNT(*) FROM EncryptedLogEntity 
+        WHERE uploadStateDbValue = :uploadState
+    """)
+    internal abstract suspend fun getEncryptedLogsCount(uploadState: Int): Int
+
+    @Insert
+    internal abstract suspend fun insertEncryptedLog(encryptedLog: EncryptedLogModel)
```

Comment:
This is only used in this DAO, let's make it not accessible outside if it's not necessary 
```suggestion
    protected abstract suspend fun insertEncryptedLog(encryptedLog: EncryptedLogModel)
```

Reply:
Thanks @wzieba, this is only used in this DAO because of the helper functions ([related](https://github.com/Automattic/EncryptedLogging/pull/20#discussion_r2163819958)), plus used in tests, I would rather keep this as `internal` for now. 🤔

PS: [Same](https://github.com/Automattic/EncryptedLogging/pull/20#discussion_r2163814097)

Reply:
Related: 28502efa3064ed0019d3e384d6dc3f0364e014b0

Reply:
Sure - as helper functions are removed and this method is used in Store, then we can't make it dao-only.